buildpack_integration_test.go

```diff
@@ -0,0 +1,199 @@
+package acceptance_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/paketo-buildpacks/occam"
+	. "github.com/paketo-buildpacks/occam/matchers"
+	"github.com/paketo-buildpacks/packit/v2/pexec"
+)
+
+func testBuildpackIntegration(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect     = NewWithT(t).Expect
+		Eventually = NewWithT(t).Eventually
+
+		buildPlanBuildpack string
+		goDistBuildpack    string
+
+		builderConfigFilepath string
+
+		pack    occam.Pack
+		docker  occam.Docker
+		source  string
+		name    string
+		builder string
+
+		image     occam.Image
+		container occam.Container
+	)
+
+	it.Before(func() {
+		pack = occam.NewPack().WithVerbose()
+		docker = occam.NewDocker()
+
+		var err error
+		name, err = occam.RandomName()
+		Expect(err).NotTo(HaveOccurred())
+
+		buildpackStore := occam.NewBuildpackStore()
+
+		buildPlanBuildpack, err = buildpackStore.Get.
+			Execute("github.com/paketo-community/build-plan")
+		Expect(err).NotTo(HaveOccurred())
+
+		goDistBuildpack, err = buildpackStore.Get.
+			WithVersion("1.2.3").
+			Execute("github.com/paketo-buildpacks/go-dist")
+		Expect(err).NotTo(HaveOccurred())
+
+		source, err = occam.Source(filepath.Join("integration", "testdata", "simple_app"))
+		Expect(err).NotTo(HaveOccurred())
+
+		builderConfigFile, err := os.CreateTemp("", "builder.toml")
+		Expect(err).NotTo(HaveOccurred())
+		builderConfigFilepath = builderConfigFile.Name()
+
+		_, err = fmt.Fprintf(builderConfigFile, `
+[stack]
+  build-image = "%s:latest"
+  id = "io.buildpacks.stacks.ubi8"
+  run-image = "%s:latest"
+`,
```

Comment:
it seems like this test is only gong to test out the basic run image scenario. Are you planning to add any sort of test for the other run images?

Reply:
I left that out of this initial set of testing as I think we likely want some testing of the runtime in the run images (in the initial case of the Node.js runtime), but that requires us having a working published extension, which in turn requires this stack to be published. 

I could add the basic testing which is done for the default build/run but I was thinking it would be better to wait until we can do a test that validates the basic functionality of the runtime in the run images.

Reply:
@mhdawson can we capture this in an issue?

Reply:
@sophiewigmore created here - https://github.com/paketo-community/ubi-base-stack/issues/6